runtime/capsules/runtime/src/mctp/mctp_binding.rs

```diff
@@ -0,0 +1,204 @@
+pub const MCTP_I3C_MAXBUF: usize = 69; // 4 MCTP header + 64 baseline payload + 1 (PEC)
+
+pub const MCTP_I3C_MAXMTU: usize = MCTP_I3C_MAXBUF - 1; // 64 bytes
+pub const MCTP_I3C_MINMTU: usize = 4 + 64; // 4 MCPT header + 64 baseline payload
+
+/// This trait contains the interface definition
+/// for sending the MCTP packet through MCTP transport binding layer.
+pub trait MCTPTransportBinding<'a> {
+    /// Set the client that will be called when the packet is transmitted.
+    fn set_tx_client(&self, client: &'a dyn TxClient);
+
+    /// Set the client that will be called when the packet is received.
+    fn set_rx_client(&self, client: &'a dyn RxClient);
+
+    /// Set the buffer that will be used for receiving packets.
+    fn set_rx_buffer(&self, rx_buf: &'static mut [u8]);
+
+    fn transmit(&self, tx_buffer: &'static mut [u8]) -> Result<(), (ErrorCode, &'static mut [u8])>;
+
+    /// Enable/Disable the I3C target device
+    fn enable(&self);
+    fn disable(&self);
+
+    /// Get the maximum transmission unit (MTU) size.
+    fn get_mtu_size(&self) -> usize;
+
+    /// Get hdr size of transport binding layer
+    fn get_hdr_size(&self) -> usize;
+}
+
+pub struct MCTPI3CBinding<'a> {
+    /// Reference to the I3C Target device driver.
+    i3c_target: &'a dyn I3CTarget<'a>,
+    rx_client: OptionalCell<&'a dyn RxClient>,
+    tx_client: OptionalCell<&'a dyn TxClient>,
+    /// I3C Target device address needed for PEC calculation.
+    device_address: Cell<u8>,
+    /// Max Read length supported by the I3C target device.
+    max_read_len: Cell<usize>,
+    /// Max Write length supported by the I3C target device.
+    max_write_len: Cell<usize>,
+    /// Buffer to store the transmitted packet.
+    tx_buffer: TakeCell<'static, [u8]>,
+}
+
+impl<'a> MCTPI3CBinding<'a> {
+    pub fn new(i3c_target: &'a dyn I3CTarget<'a>) -> MCTPI3CBinding<'a> {
+        MCTPI3CBinding {
+            i3c_target,
+            rx_client: OptionalCell::empty(),
+            tx_client: OptionalCell::empty(),
+            device_address: Cell::new(0),
+            max_read_len: Cell::new(0),
+            max_write_len: Cell::new(0),
+            tx_buffer: TakeCell::empty(),
+        }
+    }
+
+    pub fn mctp_i3c_setup(&self) {
+        let device_info = self.i3c_target.get_device_info();
+        self.max_read_len.set(device_info.max_read_len);
+        self.max_write_len.set(device_info.max_write_len);
+        self.device_address.set(
+            device_info
+                .dynamic_addr
+                .unwrap_or(device_info.static_addr.unwrap_or(0)),
+        );
+        self.i3c_target.enable();
+    }
+
+    /// SMBus CRC8 calculation.
+    fn compute_pec(addr: u8, buf: &[u8], len: usize) -> u8 {
+        let mut crc = 0u8;
+
+        crc = MCTPI3CBinding::crc8(crc, addr);
+
+        for byte in buf.iter().take(len) {
+            crc = MCTPI3CBinding::crc8(crc, *byte);
+        }
+        crc
+    }
+
+    fn crc8(crc: u8, data: u8) -> u8 {
```

Comment:
I would definitely like to see some unit tests for this. If there are SMBus test vectors, that would be ideal.

Reply:
added unit tests for PEC calculation with some dummy MCTP over I3C packets